### PR TITLE
Add isPrAlreadyOpen method

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
Add function to prevent the bumper script to open multiple bumping PRs for same release.

- **move isBumpNeeded to component_commands.go**
We want to add a new function isPrAlreadyOpen to the isBumpNeeded function.
This new function will require githubApi access, so there is no longer
a reason to keep isBumpNeeded outside component_commands.go.
This commit simply moves the functions and unit tests to the appropriate place in component_commands.go.

- **adding new function isPrAlreadyOpen to isBumpNeeded**
We want to avoid opening multiple bump PRs to the same component's new release.
Let's add a new function to check that there is no already open PR
with the same bump title.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by phoracek — CHANGES_REQUESTED** [review]
Please make the first commit so it just moves the code, no other changes should be done there. It does not need to compile. I want to see a clear diff of what was changed compared to the original.

**Comment by RamLavi** [comment]
> Please make the first commit so it just moves the code, no other changes should be done there. It does not need to compile. I want to see a clear diff of what was changed compared to the original.

Done

**Comment by RamLavi** [comment]
/retest

teir1 KMP issue fixed in [PR](https://github.com/kubevirt/cluster-network-addons-operator/pull/521). retesting.

File: tools/bumper/bumper.go [hunk 1]
```diff
@@ -97,7 +95,7 @@ func main() {
 			}
 
 			// create a new branch name
-			BranchName := strings.Replace(strings.ToLower(prTitle), " ", "_", -1)
+			BranchName := strings.Replace(strings.ToLower(proposedPrTitle), " ", "_", -1)
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply tools/bumper/bumper.go:98]: Done

File: tools/bumper/component_commands.go [hunk 2]
```diff
@@ -217,14 +227,24 @@ func (componentOps *gitComponent) getLatestFromBranch(repo, owner, branch, repoD
...
 		return false, nil
 	}
 
+	// check if PR not already opened
+	isAlreadyOpened, err := componentOps.isPrAlreadyOpened(cnaoOwner, cnaoRepo, proposedPrTitle)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to check if PR already open")
+	}
+	if isAlreadyOpened {
+		logger.Printf("Bump PR for the Latest version already exist. aborting auto bump")
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply tools/bumper/component_commands.go:244]: Done

File: tools/bumper/component_commands_test.go [hunk 3]
```diff
@@ -215,13 +215,53 @@ var _ = Describe("Testing internal git", func() {
...
 		}),
 	)
+
+	type isPrAlreadyOpenedParams struct {
+		prTitle      string
+		expectResult bool
+	}
+	DescribeTable("getUpdatedReleaseInfo function",
+		func(r isPrAlreadyOpenedParams) {
+			defer os.RemoveAll(gitComponent.gitRepo.localDir)
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply tools/bumper/component_commands_test.go:225]: not sure I understand..

File: tools/bumper/git_fakes.go [hunk 4]
```diff
@@ -52,6 +52,21 @@ func (g mockGithubApi) GetRef(owner string, repo string, ref string) (*github.Re
 	return githubRef, nil, err
 }
 
+func (g mockGithubApi) ListPullRequests(owner string, repo string) ([]*github.PullRequest, *github.Response, error) {
+	var fakePrList []*github.PullRequest
+
+	prTitleBump1 := "bump test-component to 0.0.2"
+	fakePrList = append(fakePrList, &github.PullRequest{Title: &prTitleBump1})
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply tools/bumper/git_fakes.go:59]: I agree. will change it  to something better

**Comment by RamLavi** [comment]
/hold

Still work in progress

**Comment by RamLavi** [comment]
/hold cancel

ready for review

**Comment by alonSadan** [comment]
Thank you Ram.
/lgtm

File: tools/bumper/component_commands_test.go [hunk 5]
```diff
@@ -325,62 +389,71 @@ var _ = Describe("Testing internal git", func() {
 			currentReleaseVersion: "v2.5.1",
 			latestReleaseVersion:  "v3.6.2",
 			updatePolicy:          "static",
+			prTitle:               "dummy new PR title",
```
Comment (alonSadan) [inline tools/bumper/component_commands_test.go:392]: nit: I think a const can be used here

Comment (RamLavi) [reply tools/bumper/component_commands_test.go]: [Done](https://github.com/kubevirt/cluster-network-addons-operator/compare/925a49e92daed1a59942412539cfd5a20cb05601..62c810a5cec404a8115a802800c2aab5c0f87ed7)

**Comment by alonSadan** [comment]
Great!
/lgtm

**Review by phoracek — COMMENTED** [review]
/approve

**Pull request merged**